internal/pkg/agent/application/managed_mode.go

```diff
@@ -141,8 +141,12 @@ func (m *managedConfigManager) Run(ctx context.Context) error {
 		// TODO(ph) We will need an improvement on fleet, if there is an error while dispatching a
 		// persisted action on disk we should be able to ask Fleet to get the latest configuration.
 		// But at the moment this is not possible because the policy change was acked.
-		if err := store.ReplayActions(ctx, m.log, actionDispatcher, actionAcker, actions...); err != nil {
+		m.log.Info("restoring current policy from disk")
+		actionDispatcher.Dispatch(ctx, actionAcker, actions...)
+		select {
+		case err := <-actionDispatcher.Errors():
 			m.log.Errorf("could not recover state, error %+v, skipping...", err)
+		default:
```

Comment:
I believe this whole block of code is blocking. `actionDispatcher.Dispatch` is a blocking call, its not running in a goroutine so that means the whole content of the function needs to complete before running. But if an error happens inside of the code block it will try to write to the `errCh` which will block because nothing is reading from it.

Reply:
good catch, i'll make the changes